### PR TITLE
Moved some hud menus to tablet

### PR DESCRIPTION
**Pull request by druiz17**
Moved switch, bubble, edit, goto hud menu items to the tablet

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7251fa6c18f559e508ea39f3d82a9593bc95b966 available at: https://gist.github.com/235898062b87ea1841422e6de97cab81<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-7251fa6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-7251fa6.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 74092d87140364200a2784c76e440d180fc6b1ae available at: https://gist.github.com/235898062b87ea1841422e6de97cab81<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-74092d8.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-74092d8.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: fe4fa351d131c1104a37a2f5e5e05d020faed04d available at: https://gist.github.com/235898062b87ea1841422e6de97cab81<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-fe4fa35.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-fe4fa35.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 31decffb31d848d896f125afe2f17f1ce98a1c3d available at: https://gist.github.com/235898062b87ea1841422e6de97cab81<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-31decff.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-31decff.dmg

File: scripts/system/bubble.js [hunk 1]
```diff
@@ -91,9 +91,9 @@
 
     // Used to set the state of the bubble HUD button
     function writeButtonProperties(parameter) {
-        button.writeProperty('buttonState', parameter ? 0 : 1);
-        button.writeProperty('defaultState', parameter ? 0 : 1);
-        button.writeProperty('hoverState', parameter ? 2 : 3);
+       var buttonProperties = button.getProperties();
+       buttonProperties.isActive = parameter;
+       button.editProperties(buttonProperties);
```
Comment (hyperlogic) [inline scripts/system/bubble.js]: For this you don't have to getProperties if you are only setting one property.  You can pass an object with a single field, `{isActive: parameter}`, to button.editProperties.

File: scripts/system/edit.js [hunk 2]
```diff
@@ -171,7 +171,8 @@ var toolBar = (function () {
     var that = {},
         toolBar,
```
Comment (hyperlogic) [inline scripts/system/edit.js]: should probably remove `toolBar`

File: scripts/system/edit.js [hunk 3]
```diff
@@ -231,15 +232,11 @@ var toolBar = (function () {
...
-            visible: true,
-            alpha: 0.9,
-            buttonState: 1,
-            hoverState: 3,
-            defaultState: 1
-        });
+        tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
+        activeButton = tablet.addButton({
+            text: "EDIT"
+            });
```
Comment (hyperlogic) [inline scripts/system/edit.js]: remove 4 spaces before the closing `});`

File: scripts/system/edit.js [hunk 4]
```diff
@@ -440,9 +437,9 @@ var toolBar = (function () {
 
     that.toggle = function () {
         that.setActive(!isActive);
-        activeButton.writeProperty("buttonState", isActive ? 0 : 1);
-        activeButton.writeProperty("defaultState", isActive ? 0 : 1);
-        activeButton.writeProperty("hoverState", isActive ? 2 : 3);
+        var buttonProperties = activeButton.getProperties();
+        buttonProperties.isActive = isActive;
+        activeButton.editProperties(buttonProperties);
```
Comment (hyperlogic) [inline scripts/system/edit.js]: Similar to bubble.js, just pass an object with a single field `isActive`.

File: scripts/system/goto.js [hunk 5]
```diff
@@ -13,25 +13,23 @@
 
 (function() { // BEGIN LOCAL_SCOPE
 
-var toolBar = Toolbars.getToolbar("com.highfidelity.interface.toolbar.system");
+var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
+var isActive = false
+var button = tablet.addButton({
+    text:"GOTO"})
```
Comment (hyperlogic) [inline scripts/system/goto.js]: needs semi-colons.  Also put `});` on it's own line.

File: scripts/system/goto.js [hunk 6]
```diff
@@ -13,25 +13,23 @@
...
-});
-
 function onAddressBarShown(visible) {
-    button.writeProperty('buttonState', visible ? 0 : 1);
-    button.writeProperty('defaultState', visible ? 0 : 1);
-    button.writeProperty('hoverState', visible ? 2 : 3);
+}
+
+function setActive(active) {
+  isActive = active;
```
Comment (hyperlogic) [inline scripts/system/goto.js]: 4 space indents.

File: scripts/system/goto.js [hunk 7]
```diff
@@ -13,25 +13,23 @@
...
 }
 function onClicked(){
+    setActive(!isActive);
+    var buttonProperties = button.getProperties();
+    buttonProperties.isActive = isActive;
+    button.editProperties(buttonProperties);
```
Comment (hyperlogic) [inline scripts/system/goto.js]: Similar to bubble.js you don't need to call getProperties here.

File: scripts/system/hmd.js [hunk 8]
```diff
@@ -35,15 +35,12 @@ function updateControllerDisplay() {
...
-var toolBar = Toolbars.getToolbar("com.highfidelity.interface.toolbar.system");
+var tablet = Tablet.getTablet("com.highfidelity.interface.tablet.system");
 var button;
 // Independent and Entity mode make people sick. Third Person and Mirror have traps that we need to work through.
 // Disable them in hmd.
 var desktopOnlyViews = ['Mirror', 'Independent Mode', 'Entity Mode'];
 function onHmdChanged(isHmd) {
-    button.writeProperty('buttonState', isHmd ? 0 : 1);
-    button.writeProperty('defaultState', isHmd ? 0 : 1);
-    button.writeProperty('hoverState', isHmd ? 2 : 3);
```
Comment (hyperlogic) [inline scripts/system/hmd.js]: You removed these writeProperties, I suppose we need to change the icon here...  Can you leave a TODO comment here so we don't forget it?

File: scripts/system/hmd.js [hunk 9]
```diff
@@ -54,23 +51,17 @@ function onClicked(){
...
-    button = toolBar.addButton({
-        objectName: "hmdToggle",
-        imageURL: Script.resolvePath("assets/images/tools/switch.svg"),
-        visible: true,
-        hoverState: 2,
-        defaultState: 0,
-        alpha: 0.9
-    });
+    button = tablet.addButton({
+        text: "SWITCH"});
```
Comment (hyperlogic) [inline scripts/system/hmd.js]: put `});` on its own line.

**Comment by hifi-gustavo** [comment]
Build console output for commit: bee6e230aeae32d07d11eedc251e7949298133d2 available at: https://gist.github.com/235898062b87ea1841422e6de97cab81<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-bee6e23.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9241-bee6e23.dmg

**Pull request merged**